clrtrust

```diff
@@ -84,6 +84,143 @@ find_all_certs() {
     find_certs $CLR_LOCAL_TRUST_SRC/trusted
 }
 
+print_check_help() {
+    cat <<EOF
+Usage: ${BASENAME} check [-h|--help] [-v|--verbose]
+
+    Checks the environment and reports issues found.
+
+    -h | --help         Prints this help message and exits
+    -v | --verbose      Provides extra explanations for the errors encountered
+EOF
+}
+
+cmd_check() {
+    local opt_verbose
+    local ret # return value
+
+    ret=0
+
+    while [ $# -gt 0 ]; do
+        case $1 in
+            -h|--help)
+                print_check_help
+                return 0
+                ;;
+            -v|--verbose)
+                opt_verbose=1
+                shift
+                ;;
+            *)
+                print_check_help
+                return $EINVAL
+                ;;
+        esac
+    done
+
+    if [ ! -z "${VERBOSE}" ]; then
+        opt_verbose=1
+    fi
+
+    if [ -e "${CLR_LOCAL_TRUST_SRC}" ]; then
+        if [ ! -d "${CLR_LOCAL_TRUST_SRC}" ]; then
+            1>&2 echo "${CLR_LOCAL_TRUST_SRC} must be a directory."
+            test ! -z $opt_verbose && 1>&2 cat <<EOF
+    ${CLR_LOCAL_TRUST_SRC} is used by ${BASENAME} to store local (user-added) trust and
+    distrust data.
+EOF
+            ret=$EBADST
+        fi
+        if [ -e "${CLR_LOCAL_TRUST_SRC}/distrusted" -a ! -d "${CLR_LOCAL_TRUST_SRC}/distrusted" ]; then
+            1>&2 echo "$CLR_LOCAL_TRUST_SRC/distrusted must be a directory."
+            test ! -z $opt_verbose && 1>&2 cat <<EOF
+    ${CLR_LOCAL_TRUST_SRC}/trusted is a directory used by ${BASENAME} to store
+    trusted root CA certificates.
+EOF
+            ret=$EBADST
+        fi
+        if [ -e "${CLR_LOCAL_TRUST_SRC}/trusted" -a ! -d "${CLR_LOCAL_TRUST_SRC}/trusted" ]; then
+            1>&2 echo "$CLR_LOCAL_TRUST_SRC/trusted must be a directory."
+            test ! -z $opt_verbose && 1>&2 cat <<EOF
+    ${CLR_LOCAL_TRUST_SRC}/distrusted is a directory used by ${BASENAME} to store
+    distrusted root CA certificates.
+EOF
+            ret=$EBADST
+        fi
+    fi # it's OK if CLR_LOCAL_TRUST_SRC does not exist
+
+    if [ -e "${CLR_CLEAR_TRUST_SRC}" ]; then
+        for dir in "${CLR_CLEAR_TRUST_SRC}" "${CLR_CLEAR_TRUST_SRC}/trusted"; do
+            if [ ! -d "${dir}" ]; then
+                1>&2 echo "${dir} must be a directory."
+                test ! -z $opt_verbose && 1>&2 cat <<EOF
+    ${dir} is a directory which contains trust and distrust data provided by
+    Clear Linux. It is installed initially and updated by swupd. Remove ${dir}
+    and run 'swupd verify --fix' to reinstall.
+EOF
+                ret=$EBADST
+            fi
+        done
+    else # it's not OK if CLR_CLEAR_TRUST_SRC does not exist
+        1>&2 echo "${CLR_CLEAR_TRUST_SRC} does not exist."
```

Comment:
I agree. I couldn't decide on a uniform strategy: "print all errors and then return" or "print first error and return". I like when tools do the former, but in this case it will only contaminate the logic after. Anyways, because I couldn't decide internally, it lead to this type of bugs. I think the best strategy would be to return after a series of similar-type checks. So I'll do that.